src/Nethermind/Nethermind.Evm/VirtualMachine.cs

```diff
@@ -229,7 +234,7 @@ public TransactionSubstate Run<TTracingActions>(EvmState state, IWorldState worl
                     if (typeof(TTracingActions) == typeof(IsTracing) && !currentState.IsContinuation)
                     {
                         _txTracer.ReportAction(currentState.GasAvailable, currentState.Env.Value, currentState.From, currentState.To, currentState.ExecutionType.IsAnyCreate() ? currentState.Env.CodeInfo.MachineCode : currentState.Env.InputData, currentState.ExecutionType);
-                        if (_txTracer.IsTracingCode) _txTracer.ReportByteCode(currentState.Env.CodeInfo.MachineCode);
+                        if (_txTracer.IsTracingCode) _txTracer.ReportByteCode(currentState.Env.CodeInfo.MachineCode.ToArray());
```

Comment:
Maybe lets change tracers so they accept `ReadOnlyMemory`?

Reply:
Not sure code ReadOnlyMemory would be necssarily valid later as can be a chunk on stack, so the trace would still need to `.ToArray` it

Reply:
Depends what the tracer wants to do with it. Maybe it wants only to tostring it, which it could do from ReadOnlyMemory